code/builders/builder-vite/src/codegen-modern-iframe-script.ts

```diff
@@ -66,10 +66,10 @@ export async function generateModernIframeScriptCode(options: Options, projectRo
   
     ${getPreviewAnnotationsFunction}
 
-    window.__STORYBOOK_PREVIEW__ = window.__STORYBOOK_PREVIEW__ || new PreviewWeb();
+    window.__STORYBOOK_PREVIEW__ = window.__STORYBOOK_PREVIEW__ || new PreviewWeb(importFn, getProjectAnnotations);
     
     window.__STORYBOOK_STORY_STORE__ = window.__STORYBOOK_STORY_STORE__ || window.__STORYBOOK_PREVIEW__.storyStore;
-    window.__STORYBOOK_PREVIEW__.initialize({ importFn, getProjectAnnotations });
+    window.__STORYBOOK_PREVIEW__.initialize();
```

Comment:
I wonder if the preview should just initialize itself from the constructor? Constructors cannot be async, but we don't await the init function in these files anyway (we do in tests, but we could await the `initializationPromise` there).

WDTY @ndelangen ?

Reply:
Yes, It should.

We could add a `.ready()` method that returns the promise, we do a similar enough thing for the channel.

Reply:
Oh good idea, OK I'll make those changes.

Reply:
@ndelangen done: 21fbfebd13

Do you think we should keep `initialize()` as an alias for `ready()` for back compat?

Reply:
Why keep backwards compatibility? what's using this API other than storybook internals?

Reply:
Well React Native for one (although they will need to adjust anyway). 

I'm not aware of anyone else that would use it. They would need to adjust their calling of `new Preview()` anyways, so I guess there's no point.